app/lib/actions/candlepin/owner/upstream_regenerate_certificates.rb

```diff
@@ -0,0 +1,17 @@
+module Actions
+  module Candlepin
+    module Owner
+      class UpstreamRegenerateCertificates < Candlepin::Abstract
+        input_format do
+          param :organization_id
+          param :upstream
+        end
+
+        def run
+          organization = ::Organization.find(input[:organization_id])
+          output[:response] = organization.redhat_provider.owner_regenerate_upstream_certificates(input[:upstream])
```

Comment:
It is async, but the way candlepin works is it just marks the objects as dirty and does the actual cert generation when updated objects are requested. Here is a compare timings with Cert generation and without -> https://partha.fedorapeople.org/scratch/withCert.png vs https://partha.fedorapeople.org/scratch/noCert.png 
Note the timings for UpstreamExport is not that different. So I believe cert regeneration shouldn't take long.